sdk/rust/oak/src/lib.rs

```diff
@@ -539,3 +539,36 @@ macro_rules! entrypoint {
         }
     };
 }
+
+/// Similar to [`entrypoint`], but offers a less verbose API in case in which the handler is a
+/// [`CommandHandler`] instance. In this case, the type of incoming messages is inferred from the
+/// generic type associated with the handler instance, and only the entrypoint name needs to be
+/// specified.
+///
+/// The generated implementation of the body of this node sets up logging, and then immediately
+/// calls [`run_command_loop`] on the provided handler.
+///
+/// ```
+/// use oak_abi::proto::oak::application::ConfigMap;
+///
+/// struct Main;
+///
+/// impl oak::CommandHandler<ConfigMap> for Main {
+///     fn handle_command(&mut self, command: ConfigMap) -> anyhow::Result<()> {
+///         // ...
+/// #        unimplemented!()
+///     }
+/// }
+///
+/// oak::entrypoint_command_handler!(oak_main => Main);
```

Comment:
I still think that having a regular `main` function will be more obvious and dev friendly rather then having to define multiple structs, like `Main` and `Router` in addition to structs for all the nodes.

And also devs need to treat these new structs as Nodes (since they are created via `node_create`), which increases the number of nodes even in the simple application.

Reply:
Do you mean just for the main node, make that a function instead of a struct? My point is that, since it really is a node, we should treat it as a node. I guess it's like in Java even main is inside a class, since everything is a class. Also I think that the planner node @conradgrobler is exploring works better if there is already an assumption that the entry point looks like a node (which it is). Basically I think special casing the main node and make it look different from other nodes, when it actually is the same, is actually more confusing.